presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -233,14 +288,36 @@ public Type getType()
 
         public void processPage(Page page)
         {
-            if (step.isInputRaw()) {
+            if (!this.orderByChannels.isEmpty()) {
+                checkArgument(step == Step.SINGLE && pageIndex.isPresent(), "Order by over distributed aggregation is not supported");
+                pageIndex.ifPresent(value -> value.addPage(page));
```

Comment:
Ya, but IntelliJ still gives warning and the looks a little annoying, presumably because the check happens in another function?